src/higley_lab_to_nwb/extractors/processed_imaging_extractor.py

```diff
@@ -0,0 +1,169 @@
+            Accepted values: "dff_final", "dff_blue", "dff_uv", "blue", "uv",  "green".
+        Notes
+        -----
+        """
+        super().__init__()
+        self.file_path = Path(file_path)
+        self._sampling_frequency = sampling_frequency
+        self._times = None
+
+        self._channel_names = process_type
+
+        if process_type in ["dff_final", "dff_blue", "dff_uv"]:
+            self._pixels_matrix, self._num_rows, self._num_columns = _load_data_from_dual_imaging_configuration(
+                file_path=file_path, process_type=process_type
+            )
+            mask = _get_mask_from_dual_imaging_configuration(file_path=file_path)
+        elif process_type in ["blue", "uv", "green"]:
+            self._pixels_matrix, self._num_rows, self._num_columns = _load_data_from_1p_imaging_configuration(
+                file_path=file_path, process_type=process_type
+            )
+
+        self._num_frames, number_of_pixels = self._pixels_matrix.shape
+
+        if number_of_pixels == self._num_rows * self._num_columns:
+            self._mask = None
+
+        elif number_of_pixels == np.count_nonzero(mask):
+            self._mask = mask
+
+        else:
+            raise ValueError(
+                f"Can't reconstruct frame from pixel matrix. The total number of pixels {number_of_pixels} does "
+                f"not match the frame dimension {self._num_rows} * {self._num_columns} or the non-zero elements in "
+                f"the frame mask {np.count_nonzero(mask)}."
+            )
+
+    def _get_single_frame(self, frame_idx: int) -> np.ndarray:
+        frame = np.zeros((self._num_rows, self._num_columns))
+        i = 0
+        for r in range(self._num_rows):
+            for c in range(self._num_columns):
+                # "dff_blue" and "dff_uv" pixel matrix contains the pixel trace for only the non-zero elements of the
+                # mask (different pixel indexing)
+                if self._mask is not None and self._mask[r, c] != 0:
+                    frame[r, c] = self._pixels_matrix[frame_idx, i]
+                    i += 1
+                elif self._mask is None:
+                    frame[r, c] = self._pixels_matrix[frame_idx, i]
+                    i += 1
+        return frame
+
+    def get_video(self, start_frame=None, end_frame=None, channel: int = 0) -> np.ndarray:
+        """Get the video frames.
+
+        Parameters
+        ----------
+        start_frame: int, optional
+            Start frame index (inclusive).
+        end_frame: int, optional
+            End frame index (exclusive).
+
+        Returns
+        -------
+        video: numpy.ndarray
+            The video frames.
+        """
+
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self._num_frames
+        num_frames = end_frame - start_frame
+
+        video = np.zeros((num_frames, self._num_rows, self._num_columns))
```

Comment:
Can you double check for me if the dtype of the pixels_matrix that you're reading in still matches the dtype of the video in the nwbfile?

Reply:
Yep they do
```
extractor.get_dtype()
Out[14]: dtype('float64')
extractor._pixels_matrix[0].dtype
Out[15]: dtype('float64')
video_processed[0,:,:].dtype
Out[16]: dtype('float64')
```